### PR TITLE
@task PDA-1192 Remove item stock from item table and move related operations to inventory service

### PR DESCRIPTION
**Pull request by tren-parasoft**
Remove item stock from item table and move related operations to inventory service

### CODE REVIEW

**Review by rwang-parasoft — CHANGES_REQUESTED** [review]
Need to modify code.

File: src/main/java/com/parasoft/demoapp/service/ItemInventoryService.java [hunk 1]
```diff
@@ -109,4 +109,22 @@ private String validateItemStockBeforeIncrease(List<InventoryInfoDTO> requestedI
         }
         return "";
     }
+
```
Comment (rwang-parasoft) [inline src/main/java/com/parasoft/demoapp/service/ItemInventoryService.java:112]: All changes need to add validations.

File: src/main/java/com/parasoft/demoapp/service/OrderService.java [hunk 2]
```diff
@@ -39,6 +39,9 @@ public class OrderService {
     @Autowired
     private LocationService locationService;
 
+    @Autowired
+    private ItemInventoryService itemInventoryService;
```
Comment (rwang-parasoft) [inline src/main/java/com/parasoft/demoapp/service/OrderService.java:43]: Introduce the itemInventoryService but doesn't use it, need to double check.

File: src/main/java/com/parasoft/demoapp/service/ItemService.java [hunk 3]
```diff
@@ -70,10 +75,13 @@ public ItemEntity addNewItem(
         }
         
         ItemEntity item = new ItemEntity(name, description, categoryId, inStock, imagePath, region, new Date());
+        ItemEntity result = itemRepository.save(item);
+        result.setInStock(itemInventoryService.saveItemInStock(result.getId(), result.getInStock()).getInStock());
```
Comment (yshi-parasoft) [inline src/main/java/com/parasoft/demoapp/service/ItemService.java:79]: "result" should already have the "inStock" property and no need to set again.

**Review by rwang-parasoft — APPROVED** [review]

**Pull request merged**